### DISCUSSION TITLE
Looking for sample that modifies a response

### DISCUSSION THREAD

**Discussion by brian2github** [dotnet/yarp#270] [General] [answered]
### Some details

I have built the sample apps, and I thought the reverse proxy sample would provide an example of modifying a response.

Can you please point out how I would modify a response while acting as a proxy, and if there is a sample that does this?

Thank you.

**Comment by Tratcher** [comment, marked as answer]
What aspect of the response?

Preview2 added support for modifying response headers:
https://microsoft.github.io/reverse-proxy/articles/transforms.html#response-headers-and-trailers

Modifying the response body is not a feature we plan first class support for in the proxy, it's very content dependent. However, it is possible to do this via your own middleware. Here's one example:
https://github.com/dotnet/aspnetcore/blob/04ecd5539f7476ea91c504fea7aa765d7df66e6b/src/Middleware/ResponseCompression/src/ResponseCompressionMiddleware.cs
